### PR TITLE
Core: Add alias for glue, nessie, snowflake, jdbc catalog

### PR DESCRIPTION
**Pull request by zhaomin1423**
Add an alias to make use easier.

### CODE REVIEW

**Comment by dramaticlly** [comment]
looks like spotless failed, can you run `./gradlew :iceberg-core:spotlessApply` to fix the style error?

File: core/src/main/java/org/apache/iceberg/CatalogUtil.java [hunk 1]
```diff
@@ -67,10 +67,18 @@ public class CatalogUtil {
   public static final String ICEBERG_CATALOG_TYPE_HADOOP = "hadoop";
   public static final String ICEBERG_CATALOG_TYPE_HIVE = "hive";
   public static final String ICEBERG_CATALOG_TYPE_REST = "rest";
+  public static final String ICEBERG_CATALOG_TYPE_GLUE = "glue";
```
Comment (ajantha-bhat) [inline core/src/main/java/org/apache/iceberg/CatalogUtil.java:70]: Nice. Can we also update the docs and some testcases to use this short form?

Comment (nastra) [reply core/src/main/java/org/apache/iceberg/CatalogUtil.java:70]: +1

**Comment by zhaomin1423** [comment]
> looks like spotless failed, can you run `./gradlew :iceberg-core:spotlessApply` to fix the style error?

Thanks, you are right.

File: core/src/main/java/org/apache/iceberg/CatalogUtil.java (hunk 1 above)
Comment (zhaomin1423) [reply core/src/main/java/org/apache/iceberg/CatalogUtil.java:70]: Sorry, I don't know how to build a glue, nessie and snowflake to test. What should I do?

Comment (ajantha-bhat) [reply core/src/main/java/org/apache/iceberg/CatalogUtil.java:70]: Sorry, I missed this PR. 

Existing test code, you can look up "org.apache.iceberg.nessie.NessieCatalog" and replace few of them with alias. 

And in the docs, we can replace all the occurrences with alias.

Comment (ajantha-bhat) [reply core/src/main/java/org/apache/iceberg/CatalogUtil.java:70]: Let me know if you still want to work on it. If not, I can take it up. Thanks.

**Comment by ajantha-bhat** [comment]
since there is no response, I handled the testcase and doc updates and opened a new PR with @zhaomin1423 as coauthor


closing this in favour of https://github.com/apache/iceberg/pull/9647

**Pull request closed without merging**